Fliglio/Config/VaultPropertySetProvider.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Fliglio\Config;
+
+use Fliglio\Vault\VaultClient;
+
+class VaultPropertySetProvider implements PropertySetProvider {
+
+	private $client;
+	private $projectName;
+
+	public function __construct(VaultClient $client, $projectName) {
```

Comment:
instead of passing in project name, maybe pass in the full key path: `secret/inmo-leads`